### PR TITLE
Add test cases for Admin UI MDS

### PR DESCRIPTION
**Pull request by wanglam**
### Description

This PR is for adding functional tests for Admin UI MDS. Mainly includes two changes.
1. Add `currentBackendEndpoint` to allow overwrite request default opensearch endpoint when multi data source enabled.
2. Separate Admin UI tests cases to models and remote models, add tests cases for multi data sources.

~~To avoid build error, [this commit](https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1229/commits/f69ff57c08f7ff265e0b8fc2113aba7fd9955212) temp remove notification plugin for test, need to remove this commit when merge.~~

The CI of this PR can be passed in https://github.com/opensearch-project/opensearch-dashboards-functional-test/actions/runs/8826342921/job/24231928437?pr=1229.

### Issues Resolved

[List any issues this PR will resolve]

### Check List

- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/utils/commands.js [hunk 1]
```diff
@@ -23,6 +23,41 @@ export const CURRENT_TENANT = {
...
+  return Object.freeze({
+    DEFAULT: DEFAULT_ENDPOINT,
+    REMOTE_NO_AUTH: REMOTE_NO_AUTH_ENDPOINT,
+    /**
+     * Change current backend endpoint
+     * @param {*} changedEndPoint
+     * @param {*} immediately set immediately false to change tenant after all pending promise be invoked,
+     * useful for reset backend endpoint after all tests run.
+     */
+    set(changedEndPoint, immediately = true) {
```
Comment (tianleh) [inline cypress/utils/commands.js:41]: For ease of troubleshooting, could we add a `cy.log` inside the `set` function about the information in your comment code https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1229/files#diff-aba0ffbc791fcd54335abb5eecbfa9dbb4581b4c29df4fd3f41a7d0eca4af7d8R26

**Review by tianleh — DISMISSED** [review]
LGTM

Requesting change to remove `          ./bin/opensearch-dashboards-plugin remove notificationsDashboards` to avoid accidental merge.

**Comment by SuZhou-Joe** [comment]
Convert to draft as it tagged as DO_NOT_MERGE

File: cypress/utils/commands.js (hunk 1 above)
Comment (wanglam) [reply cypress/utils/commands.js:41]: The `cy.log` has been added to the `set` function.

**Comment by wanglam** [comment]
@tianleh @SuZhou-Joe  The `./bin/opensearch-dashboards-plugin remove notificationsDashboards` has been removed, the CI has been passed in https://github.com/opensearch-project/opensearch-dashboards-functional-test/actions/runs/8826342921/job/24231928437?pr=1229 . PR now ready to merge.

**Review by ruanyl — APPROVED** [review]

**Review by tianleh — APPROVED** [review]

**Pull request merged**